### ISSUE TITLE
Feature: Visualize your picks place in the queue.

### ISSUE THREAD

**Issue by fczuardi** [lnbits/jukebox#13]
The jukebox page currently doesnt render the queue, so a person paying for including a track dont have an idea of when it is expected to play.

**Comment by fczuardi** [comment]
A possible implementation:

```diff
diff --git a/templates/jukebox/jukebox.html b/templates/jukebox/jukebox.html
index bcad502..c80e701 100644
--- a/templates/jukebox/jukebox.html
+++ b/templates/jukebox/jukebox.html
@@ -56,6 +56,31 @@
         </q-virtual-scroll>
       </q-card-section>
     </q-card>
+    <q-card class="q-mt-lg">
+      <q-card-section>
+        <p style="font-size: 22px">Upcoming</p>
+      </q-card-section>
+      <q-card-section>
+        <q-virtual-scroll
+          style="max-height: 300px"
+          :items="upcomingTracks"
+          separator
+        >
+          <template v-slot="{ item, index }">
+            <q-item
+              :key="index"
+              dense
+            >
+              <q-item-section>
+                <q-item-label>
+                  {% raw %} {{ item.name }} - ({{ item.artist }}){% endraw %}
+                </q-item-label>
+              </q-item-section>
+            </q-item>
+          </template>
+        </q-virtual-scroll>
+      </q-card-section>
+    </q-card>
 
     <q-dialog v-model="receive.dialogues.first" position="top">
       <q-card class="q-pa-lg lnbits__dialog-card">
@@ -108,6 +133,7 @@
       return {
         currentPlaylist: [],
         currentlyPlaying: {},
+        upcomingTracks: [],
         playlists: {},
         playlist: '',
         heavyList: [],
@@ -229,8 +255,11 @@
         LNbits.api
           .request('GET', '/jukebox/api/v1/jukebox/jb/currently/{{juke_id}}')
           .then(function (res) {
-            if (res.data.id) {
-              self.currentlyPlaying = res.data
+            if (res.data.current_track.id) {
+              self.currentlyPlaying = res.data.current_track
+            }
+            if (res.data.upcoming_tracks) {
+              self.upcomingTracks = res.data.upcoming_tracks
             }
           })
           .catch(function (error) {
diff --git a/views_api.py b/views_api.py
index f0a2ca5..23c7590 100644
--- a/views_api.py
+++ b/views_api.py
@@ -410,7 +410,7 @@ async def api_get_jukebox_currently(
         try:
             assert jukebox.sp_access_token
             r = await client.get(
-                "https://api.spotify.com/v1/me/player/currently-playing?market=ES",
+                "https://api.spotify.com/v1/me/player/queue?market=ES",
                 timeout=40,
                 headers={"Authorization": "Bearer " + jukebox.sp_access_token},
             )
@@ -420,14 +420,30 @@ async def api_get_jukebox_currently(
                 try:
                     response = r.json()
 
-                    track = {
-                        "id": response["item"]["id"],
-                        "name": response["item"]["name"],
-                        "album": response["item"]["album"]["name"],
-                        "artist": response["item"]["artists"][0]["name"],
-                        "image": response["item"]["album"]["images"][0]["url"],
+                    current_track = {
+                        "id": response["currently_playing"]["id"],
+                        "name": response["currently_playing"]["name"],
+                        "album": response["currently_playing"]["album"]["name"],
+                        "artist": response["currently_playing"]["artists"][0]["name"],
+                        "image": response["currently_playing"]["album"]["images"][0]["url"],
+                    }
+
+                    upcoming_tracks = [
+                        {
+                            "id": track["id"],
+                            "name": track["name"],
+                            "album": track["album"]["name"],
+                            "artist": track["artists"][0]["name"],
+                            "image": track["album"]["images"][0]["url"],
+                        }
+                        for track in response["queue"]
+                    ]
+
+                    result = {
+                        "current_track": current_track,
+                        "upcoming_tracks": upcoming_tracks
                     }
-                    return track
+                    return result
                 except:
                     raise HTTPException(
                         status_code=HTTPStatus.NOT_FOUND, detail="Something went wrong"

```

**Comment by talvasconcelos** [comment]
that approach is correct, but you can't make the call from the frontend or you'll expose the access_token

It's fixed in #16
